upstairs/src/client.rs

```diff
@@ -2197,7 +2215,7 @@ impl DownstairsClient {
     /// different connections to the same Downstairs.
     pub(crate) fn get_connection_id(&self) -> Option<u64> {
         if self.client_task.client_stop_tx.is_some() {
-            Some(self.connection_id as u64)
+            Some(self.connection_id.0)
```

Comment:
Sorry to be so pedantic, but now that you've added the new type, we should actually use it – this function should be changed to return an `Option<ConnectionId>`, and then fix all of the compiler errors and we should be good :)